converter.go

```diff
@@ -113,21 +117,21 @@ func valueToString(v driver.Value, tsmode snowflakeType) (*string, error) {
 		s := v1.String()
 		return &s, nil
 	case reflect.Struct:
-		if tm, ok := v.(time.Time); ok {
-			switch tsmode {
-			case dateType:
+		if tm, ok := v.(time.Time); ok && dataType != nil {
+			switch {
```

Comment:
this looks like a type switch (i don't see any on type case in the changed lines). If that's true, why use Equals instead of a type switch ?

Reply:
`DataTypeDate`, `DataTypeTime`, etc. are values of type `SnowflakeDataType`, not distinct types (it's an enum). Since `SnowflakeDataType` is not a primitive type, it doesn't implement `==` and so Go won't let me do a value-based switch like we have in the left-hand side of the diff.